src/arrays/referencing_array_and_its_elements/hard/q6/MainTest.java

```diff
@@ -0,0 +1,94 @@
+package arrays.referencing_array_and_its_elements.hard.q6;
+
+import global.BaseTest;
+import global.MethodTest;
+import global.exceptions.InvalidClauseException;
+import global.tools.CustomAssertions;
+import global.tools.TestOption;
+import global.variables.Clause;
+import global.variables.clauses.NewLine;
+import global.variables.clauses.PlaceHolder;
+import global.variables.clauses.StringLiteral;
+import global.variables.wrappers.Optional;
+import org.junit.jupiter.params.ParameterizedTest;
+import org.junit.jupiter.params.provider.Arguments;
+import org.junit.jupiter.params.provider.MethodSource;
+
+import java.util.Arrays;
+import java.util.stream.Stream;
+
+import static global.tools.CustomAssertions._assertEquals;
+import static global.utils.ArrayUtil.arrayToInput;
+import static global.utils.ArrayUtil.generateRandomArray;
+import static org.junit.Assert.assertEquals;
+
+public class MainTest extends BaseTest {
+    // Java
+
+    public static int n = 10;
+
+    public Clause[] testSentence() {
+        TestOption.isInputTest = true;
+        TestOption.defaultInput = "3 4 6 21 48 42 89 2 0 91";
```

Comment:
you'll need to generate an array if you have a variable to vary the amount of inputs.